sway-ir/src/optimize/mem2reg.rs

```diff
@@ -148,19 +151,63 @@ pub fn compute_livein(
     result
 }
 
-/// Promote local values that are accessed via load/store to SSA registers.
-/// We promote only locals of non-copy type, whose every use is in a `get_local`
```

Comment:
Not related to this PR, but shouldn't this comment actually be other way around, we promote only _copy types_. The `is_copy_type`s functions see copy types as those that fit into a register. As the existing comments around the `TypeInfo::is_copy_type` implies, I am not happy neither with the choice of the name nor how those functions are now used. But sticking to that terminology, the promotable types are actually "copy types" and not "non-copy types". BDW, the `memcpyopt::is_copy_type` is actually identical to `is_promotable_type`.